assets/js/admin-script.js

```diff
@@ -303,12 +263,46 @@ jQuery( function( $ ) {
 		triggerPreview();
 	} );
 
-	function settingsChanged( previewDelay ) {
+	// Check for settings change
+	$( document ).on( 'keyup paste', '#wpo-wcpdf-settings input, #wpo-wcpdf-settings textarea', settingsChanged );
+	$( document ).on( 'change', '#wpo-wcpdf-settings input[type="checkbox"], #wpo-wcpdf-settings input[type="radio"], #wpo-wcpdf-settings select', function( event ) {
+		if ( ! event.isTrigger ) { // exclude programmatic triggers that aren't actually changing anything
+			settingsChanged( event );
+		}
+	});
+	$( document ).on( 'select2:select select2:unselect', '#wpo-wcpdf-settings select.wc-enhanced-select', settingsChanged );
+	$( document.body ).on( 'wpo-wcpdf-media-upload-setting-updated', settingsChanged );
+	$( document ).on( 'click', '.wpo_remove_image_button', settingsChanged );
+	
+	function settingsChanged( event, previewDelay ) {
+
+		// Show secondary save button
 		showSaveBtn();
-		triggerPreview( previewDelay );
+
+		// Check if preview needs to reload and with what delay
+		if ( ! settingIsExcludedForPreview( $(event.target).attr('name') ) ) {
+
+			let element = event.target.nodeName;
+			let elementType = $(event.target).attr('type');
+			let elementClass = $(event.target).attr('class');
+			
+			if ( jQuery.inArray( event.type, ['keyup', 'paste'] ) !== -1 ) {
+				if ( ( element == 'INPUT' && elementType == 'checkbox' ) || ( element == 'SELECT' && elementClass == 'dropdown-add-field' ) ) {
+					return;
+				} else {
+					previewDelay = event.type == 'keyup' ? 1000 : 0; 
+				}
+			} else if ( event.type == 'change' ) {
+				if ( element == 'INPUT' && elementType == 'checkbox' ) {
+					previewDelay = 1000;
+				}
+			}
```

Comment:
what's the value of `previewDelay` for events that don't meet the above criteria, for example `click`? (does it matter that it's undefined?)

Reply:
When `previewDelay` is undefined it will become `0` in the `triggerPreview()` function.

`timeoutDuration = typeof timeoutDuration == 'number' ? timeoutDuration : 0;`

Reply:
Thanks! I do wonder if this logic is perhaps inverted. Checkboxes should trigger the preview immediately, right? Only text fields need the delay (in case someone is still typing).

Reply:
I took it from the original code. I think the idea was that users can, by accident, double click a checkbox? Maybe @alexmigf can clarify?

Reply:
double clicking would be less than a 1 second delay. The delay should only be used in case we may expect more updates (to prevent hammering), like when someone's typing in a text box.

Reply:
I'll remove the delay for checkboxes. From a UX point of view I think it makes the reload process unnecessarily slow.

Reply:
Delay for checkboxes removed in: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/355/commits/0b90e98bdf5e9e648c716900933ac017a7f05cd2